src/js/game/LevelMVC/LevelPlane.js

```diff
@@ -311,9 +344,41 @@ module.exports = class LevelPlane extends Array {
       posToRefresh.push(this.redstoneListON[i]);
     }
 
+    // Once we're done updating redstoneWire states, check to see if doors should open/close.
+    // Do things a bit different
+    if (OnDoor) {
+      this.findDoorToAnimate(this.coordinatesToIndex(DoorPosition));
+    } else {
+      this.findDoorToAnimate(-1);
+    }
```

Comment:
This definitely seems like a weird place to have this logic; any reason it shouldn't just live in the `setBlockAt` method above?

Reply:
It was only in there because the state of the door will only ever change when redstone is changing. Meaning that if getRedstone isn't called (so redstone circuits aren't changing), the doors don't even need to do work.

If anything getRedstone() is not a great function. It's more like... "Anything pertaining to redstone needs to re-evaluate its life because something might have changed".

It can likely be taken into placeBlockAt() without losing any functionality if that's more desirable. 👍

Reply:
Ahh, that makes sense. Presumably we want the doors to change when switches are flipped or when redstone is destroyed as well, right? Can I assume that a switch flip does not go through `placeBlockAt` but does go through `getRedstone`?